src/backend/storage/smgr/md.c

```diff
@@ -514,16 +514,12 @@ mdunlinkfork(RelFileNodeBackend rnode, ForkNumber forkNum, bool isRedo, char rel
 	/*
 	 * Delete or truncate the first segment.
 	 */
+
+	/* Prevent other backends' fds from holding on to the disk space */
+	ret = do_truncate(path);
```

Comment:
Why do we truncate only the first segment for the temp relation, and not truncating the rest, as it done for regular relations?